### PR TITLE
Feat/datatypes v2

### PR DESCRIPTION
**Pull request by tdeleanu**
Adding graph datatypes and device types endpoints.
Also count for resolvers

### CODE REVIEW

File: lib/Services/cayenne.js [hunk 1]
```diff
@@ -357,17 +366,98 @@ class Cayenne {
...
+  /**
+   * Get V2 one things data type properties
+   * @param {String} typeId
+   * @param {Number} propertyId
+   * @param {Object} query
+   * @param {Number} [query.limit]
+   * @param {Number} [query.page]
+   */
+  createThingDataTypePropertyV2(typeId, payload){
+    ///v2/things/datatypes/{id}/properties/{pid}
```
Comment (hugocortes) [inline lib/Services/cayenne.js:430]: nit: can you remove unnecessary comments

File: lib/Services/cayenne.js [hunk 2]
```diff
@@ -428,6 +527,19 @@ class Cayenne {
     });
   }
 
+
+  /**
+   * Count a device's meta data by device type id
+   * @param {String}  deviceTypeId
+   * @param {Objet}   [opts={}]                    Additional options
```
Comment (hugocortes) [inline lib/Services/cayenne.js:534]: `opts` is not a parameter for this method.

File: lib/Services/cayenne.js [hunk 3]
```diff
@@ -509,6 +662,33 @@ class Cayenne {
     return this.auth.send(this.service, 'POST', url, { payload });
   }
 
+  /**
+   * @param {String}  deviceTypeId
```
Comment (hugocortes) [inline lib/Services/cayenne.js:666]: `channelId` is missing from jsdocs

File: lib/Services/cayenne.js [hunk 4]
```diff
@@ -509,6 +662,33 @@ class Cayenne {
...
+   */
+  updateDeviceTypeChannel(deviceTypeId, channelId, payload) {
+    const url = `${this.url}/things/types/${deviceTypeId}/channels/${channelId}`;
+    return this.auth.send(this.service, 'PUT', url, { payload });
+  }
+
+  /**
+   * @param {String}  deviceTypeId
+   * @param {String}  channelId
+   * @param {Object}  payload
```
Comment (hugocortes) [inline lib/Services/cayenne.js:681]: `payload` is not a parameter for this function

File: lib/Services/cayenne.js [hunk 5]
```diff
@@ -520,6 +700,34 @@ class Cayenne {
...
+   */
+  updateDeviceTypeMeta(deviceTypeId, metaId, payload) {
+    const url = `${this.url}/things/types/${deviceTypeId}/meta/${metaId}`;
+    return this.auth.send(this.service, 'PUT', url, { payload });
+  }
+
+  /**
+   * @param {String}  deviceTypeId
+   * @param {String}  metaId
+   * @param {Object}  payload
```
Comment (hugocortes) [inline lib/Services/cayenne.js:720]: `payload ` is not a parameter

File: lib/Services/cayenne.js [hunk 6]
```diff
@@ -534,6 +742,34 @@ class Cayenne {
...
+   */
+  updateDeviceTypeUse(deviceTypeId, useId, payload) {
+    const url = `${this.url}/things/types/${deviceTypeId}/meta/${useId}`;
+    return this.auth.send(this.service, 'PUT', url, { payload });
+  }
+
+  /**
+   * @param {String}  deviceTypeId
+   * @param {String}  useId
+   * @param {Object}  payload
```
Comment (hugocortes) [inline lib/Services/cayenne.js:762]: `payload` is not a parameter

**Comment by tdeleanu** [comment]
> just minor nitpicks with comments and jsdocs

done

**Pull request merged**